FLT/DedekindDomain/FiniteAdeleRing/BaseChange.lean

```diff
@@ -196,28 +196,152 @@ noncomputable def adicCompletionComapSemialgHom (v : HeightOneSpectrum A) (w : H
   UniformSpace.Completion.mapSemialgHom _ <|
   IsDedekindDomain.HeightOneSpectrum.adicValued.continuous_algebraMap A K L B v w hvw
 
--- Do we even need to prove that this map is continuous? It will follow from
--- the fact that it's K_v-linear once we know L_w has the K_v-module topology,
--- which is the next lemma.
--- omit [IsIntegralClosure B A L] [FiniteDimensional K L] [Algebra.IsSeparable K L]
---     [Module.Finite A B] in
--- lemma adicCompletionComapSemialgHom_continuous
---     (v : HeightOneSpectrum A) (w : HeightOneSpectrum B) (hvw : w.comap A = v) :
---     Continuous (adicCompletionComapSemialgHom A K L B v w hvw) := by
---   convert UniformSpace.Completion.continuous_extension (β := (adicCompletion L w))
+section ModuleTopology
+
+-- Make (v.adicCompletion K) a normed field.
+-- This exists for number fields in Mathlib, but not for general Dedekind Domains.
+-- v.asIdeal.absNorm may be 0 so just use 2 as the base for the norm.
+
+open WithZeroMulInt in
+noncomputable local instance adicCompletion_RkOne :
+    Valuation.RankOne (Valued.v : Valuation (v.adicCompletion K) ℤₘ₀) where
+  hom := {
+    toFun := toNNReal (by norm_num : (2 : NNReal) ≠ 0)
+    map_zero' := rfl
+    map_one' := rfl
+    map_mul' := MonoidWithZeroHom.map_mul (toNNReal (by norm_num))
+  }
+  strictMono' := toNNReal_strictMono (by norm_num)
+  nontrivial' := by
+    rcases Submodule.exists_mem_ne_zero_of_ne_bot v.ne_bot with ⟨x, hx1, hx2⟩
+    use algebraMap A K x
+    dsimp [adicCompletion]
+    rw [valuedAdicCompletion_eq_valuation' v (algebraMap A K x)]
+    constructor
+    · simpa only [ne_eq, map_eq_zero, FaithfulSMul.algebraMap_eq_zero_iff]
+    · apply ne_of_lt
+      rw [valuation_eq_intValuationDef, intValuation_lt_one_iff_dvd]
+      exact Ideal.dvd_span_singleton.mpr hx1
+
+noncomputable local instance adicCompletion_normedField : NormedField (adicCompletion K v) :=
+  Valued.toNormedField (hv := adicCompletion_RkOne A K v)
+      (adicCompletion K v) (WithZero (Multiplicative ℤ))
```

Comment:
you should be able to have it by doing
```
open scoped Valued
```
or `open scoped NormedField`, I don't remember which

Reply:
I can't get instance resolution to pick up the RankOne instance correctly if I do that.